feat(new_homepage): introduce action buttons

elisa-a-v opened the pull request:
Closes #5360 

This PR introduces:
- A responsive row of action buttons in the homepage.
- An Alpine component for the homepage that handles the behavior of the arrow buttons that appear in smaller screens.
- A tiny refactor where `bell.svg` is now called `bell_outline.svg`.

#### Note on keyboard navigation
I removed the arrow buttons from the tab order to avoid tabbing through them, which can be confusing and is actually pretty useless (arrows are only focusable when you've already scrolled all the way: right arrow when you get to the end, left arrow when you're at the start.)

Therefore:
- To move scroll through the actions we can simply use the native behavior with the <kbd>left</kbd> and <kbd>right</kbd> arrow keys.
- To move focus we use the native <kbd>tab</kbd> and <kbd>shift</kbd>+<kbd>tab</kbd> navigation, as well as the custom <kbd>home</kbd> and <kbd>end</kbd> keys to move to the start/end.

However, I couldn't get the <kbd>home</kbd> and <kbd>end</kbd> keys to work on Safari, but it's a _really_ minor feature so unless people think it's important I'm not sure it's worth giving it much more time.

## Screenshots

<details>
<summary>Desktop</summary>

![image](https://github.com/user-attachments/assets/46fc20f5-b377-4e84-9956-0e26b998962a)

</details>


<details>
<summary>Mobile</summary>

![image](https://github.com/user-attachments/assets/f8018a04-8ea8-424d-a7bd-5b116282e62c)

</details>


## Arrow buttons demo

[Screencast from 06-28-2025 12:58:32 AM.webm](https://github.com/user-attachments/assets/429d047b-0599-449b-b455-d5bfc4c72f88)

mlissner reviewed (dismissed):
Woah. I expected about 3× the JavaScript in this one. I didn't read it, but I'm amazed how short it is. 

Two thoughts from me and I'll let somebody else review properly:

1. The buttons really feel like they're good candidates for components. Lots of repeated code.

2. I tweaked a bunch of the words because I've had some distance from the design and the old words don't make much sense to me anymore.

Thank you!

ERosendo reviewed (changes requested):
Thanks for this PR @elisa-a-v. Code looks good. One thing I spotted is that the scrollbar style hasn't been updated. I've added a minor comment on that.

elisa-a-v commented on cl/search/templates/v2_homepage.html:
```diff
@@ -13,11 +19,138 @@
+      {% svg "chevron" class="w-6 h-6 text-greyscale-500 transform rotate-90" %}
+    </button>
+
+    <div
+      x-ref="container"
+      x-on:scroll="updateButtons"
+      x-on:keydown.home.stop.prevent="focusFirst"
+      x-on:keydown.end.stop.prevent="focusLast"
+      x-on:resize.window="updateButtons"
+      role="toolbar"
+      aria-label="Homepage actions"
+      tabindex="-1"
+      class="relative w-full flex flex-row gap-2 max-w-[1400px] mx-auto overflow-auto justify-between md:p-10 px-4 py-11"
```
Oooh you're right! I thought it was a highlight for the buttons 🤦🏽 in fact, we had already [dismissed the red underline](https://github.com/freelawproject/courtlistener/issues/5360#issuecomment-3014667202) 🤪 However, what we dismissed wasn't really the red _scrollbar_. Changing the color is trivial, and it does look pretty nice and cohesive with the rest of the design, so maybe this changes things and we do want the red scrollbar.

The standard `scrollbar-color` CSS property doesn't work in some browsers though, like Safari, so the UI would look slightly different in different browsers, but I'm not sure how relevant that is. What do you think?

mlissner replied:
Safari sucks, let's do it for the ones that support it at least. :)

elisa-a-v commented:
Thank you @ERosendo for the review. I've updated the text as requested, made an icon link component, and refactored the homepage to use this new component.

> The buttons really feel like they're good candidates for components. Lots of repeated code.

Yeah I love how much cleaner the code looks now! Good call 👍🏽

ERosendo reviewed (approved):
LGTM :shipit:
